test/unit/permissions/index.spec.js

```diff
@@ -0,0 +1,77 @@
+import { can } from '../../../lib/permissions';
+
+describe('can', () => {
+  let user;
+
+  beforeEach(() => {
+    user = {
+      get: jest.fn(),
+      is: jest.fn()
+    };
+  });
+
+  test('Rejects with an error if task is unknown', () => {
+    const task = 'unknown.task';
+    const expected = 'Unknown task: unknown.task';
+    return can(user, task)
+      .catch(err => {
+        expect(err.message).toBe(expected);
+      });
+  });
```

Comment:
I think this test needs to be slightly restructured, as currently it doesn't _actually_ assert that the promise rejects, only that _if_ it does then it has the right error. A resolved promise will still pass this test.

Should be something like:

```
return can(user, task)
  .then(
     () => expect.fail('resolved', 'rejected', 'Promise should reject'),
     err => {
     ...
     }
  );
```

Reply:
👍